packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -76,6 +76,38 @@ abstract mixin class WidgetsBindingObserver {
   /// {@macro flutter.widgets.AndroidPredictiveBack}
   Future<bool> didPopRoute() => Future<bool>.value(false);
 
+  /// Called at the start of a predictive back gesture.
+  ///
+  /// Returns true if the call should be considered handled.
+  ///
+  /// Currently, this is only used on Android devices that support the
+  /// predictive back feature.
+  bool handleStartBackGesture(PredictiveBackEvent backEvent) => false;
```

Comment:
CC @maRci002 

It looks like the [docs](https://developer.android.com/reference/android/window/BackEvent) don't provide a way to uniquely identify a gesture, but I think only one can be in progress at a time. Given WidgetBinding's implementation, it would be possible to do something weird like register a new observer in the middle of a gesture.

The Android embedder doesn't do anything with this "handled" return boolean ([code](https://github.com/flutter/engine/pull/49093/files#diff-c27c7c310f7c1bc2a6e26fdc6fd5df63aab2240efad7de52a13012ab2bc771caR66)) right now, so yes it will continue to call with the "update" values.

Reply:
This thread and https://github.com/flutter/flutter/pull/141373/files#r1536307100 raises  a good point. The API is kinda awkward right now and you could get into an odd state. I wonder if it would be better to have one method to register your interest in back gesture events and if you indicate that you want to handle it, you (and only you) get all the events of the sequence?

Reply:
Changed to the single observer approach: https://github.com/flutter/flutter/commit/ac0b088ee463ad197d5dcbc3012fb07f32b680a6

Reply:
Per https://github.com/flutter/flutter/pull/141373/files#r1538384347, when the user presses the back button,  this callback is called if Android API level >= 34, otherwise the `didPopRoute` method gets called? So if the app targets all API levels both should be implemented? If that's the case should `didPopRoute` be deprecated, or we keep using that for button events and only call handleStartBackGesture when it's a gesture event?

Reply:
for button events I guess `handleUpdateBackGestureProgress` won't be called? Do those start then immediately commit?

Reply:
Talked offline but wanted to answer here too. If you're writing some custom behavior for all forms of Android system backs:

1. Predictive back
Implement all of the handle*BackGesture methods.

2. Non-predictive system back gestures (such as older Android versions or when the flag is off now)
No need to implement anything, committing the gesture will just result in a "didPopRoute". You could override that if you wanted to do something fancy.

3. The system back button
Same as #2, it will send "didPopRoute".

For a system back button on a device that supports predictive back (I believe this is rare), then handleStartBackGesture and handleCommitBackGesture will be called, and the latter will pop the route by default.